transit_odp/timetables/utils.py

```diff
@@ -234,3 +235,253 @@ def get_line_description_based_on_direction(row: pd.Series) -> str:
+    ].astype(str)
+    return df_vehicle_journey_operating["vehicle_journey_code"].unique().tolist()
+
+
+def get_df_timetable_visualiser(
+    df_vehicle_journey_operating: pd.DataFrame,
+) -> pd.DataFrame:
+    """
+    Get the dataframe containing the list of stops and the timetable details
+    with journey code as columns
+    """
+
+    if df_vehicle_journey_operating.empty:
+        return df_vehicle_journey_operating
+
+    # Keep the relevant columns of dataframe and remove the duplicates
+    columns_to_keep = [
+        "common_name",
+        "stop_sequence",
+        "vehicle_journey_code",
+        "departure_time",
+        "atco_code",
+    ]
+    df_vehicle_journey_operating = df_vehicle_journey_operating[columns_to_keep]
+    df_vehicle_journey_operating = df_vehicle_journey_operating.drop_duplicates()
+    df_vehicle_journey_operating["key"] = df_vehicle_journey_operating.apply(
+        lambda row: f"{row['common_name']}_{row['stop_sequence']}_{row['vehicle_journey_code']}",
+        axis=1,
+    )
+
+    vehicle_journey_codes_sorted = get_vehicle_journey_codes_sorted(
+        df_vehicle_journey_operating
+    )
+
+    # Extract the stops based on the stop sequence and departure time
+    df_sequence_time: pd.DataFrame = df_vehicle_journey_operating.sort_values(
+        ["stop_sequence", "departure_time"]
+    )
+    df_sequence_time = df_sequence_time[["stop_sequence", "common_name"]]
+    df_sequence_time = df_sequence_time.drop_duplicates()
+    df_sequence_time["key"] = df_sequence_time.apply(
+        lambda row: f"{row['common_name']}_{row['stop_sequence']}",
+        axis=1,
+    )
+    bus_stops = df_sequence_time["common_name"].tolist()
+
+    # Create a dict for storing the unique combination of columns data for fast retreival
+    departure_time_data = {}
+    for row in df_vehicle_journey_operating.to_dict("records"):
+        departure_time_data[row["key"]] = row["departure_time"]
+
+    stops_journey_code_time_list = []
+    for row in df_sequence_time.to_dict("records"):
+        record = {}
+        for journey_code in vehicle_journey_codes_sorted:
+            key = f"{row['key']}_{journey_code}"
+            record[journey_code] = departure_time_data.get(key, "-")
+        stops_journey_code_time_list.append(record)
+
+    df_vehicle_journey_operating = pd.DataFrame(
+        stops_journey_code_time_list, index=bus_stops
+    )
+
+    return df_vehicle_journey_operating
+
+
+def get_non_operating_vj_serviced_org(
+    target_date: str, df_serviced_org_working_days: pd.DataFrame
+) -> List:
+    """
+    Get the vehicle journeys non-operating based on the serviced organisation
+    working days
+
+    :return: List
+    Return the non-operating vehicle journey ids
+    """
+    if df_serviced_org_working_days.empty:
+        return []
+    df_serviced_org_working_days = df_serviced_org_working_days.drop_duplicates()
+    df_serviced_org_working_days.sort_values(by=["start_date"])
```

Comment:
This would not be required if the comment to identify `df_service_operating` using group by is implemented